extensions/wikia/Wall/builders/WallMessageBuilder.class.php

```diff
@@ -94,11 +94,19 @@ public function postNewMessage(): WallMessageBuilder {
 			// Permissions check are performed on article comment level by EditPage class of Mediawiki
 			// Text is matched there against Phalanx filters and all user blocks (global and local) are also checked
 			// This can cause edit to fail
-			$this->throwException( 'Failed to create article comment' );
+			$reason =  $result && (
```

Comment:
Let's consider creating a different exception class for edits that were prevented by Phalanx content block / user block and handling it in controller. This would allow us to separate handling of failed edit caused by block from failed edit caused by DB issues or invalid input, and this way we won't log every blocked edit attempt to Kibana as error.